python/sparkdl/udf/keras_image_model.py

```diff
@@ -131,20 +134,17 @@ def _serialize_and_reload_with(preprocessor):
     """
     def udf_impl(spimg):
         import numpy as np
-        from PIL import Image
         from tempfile import NamedTemporaryFile
-        from sparkdl.image.imageIO import imageArrayToStruct, imageType
+        from sparkdl.image.imageIO import imageArrayToStruct
 
-        pil_mode = imageType(spimg).pilMode
-        img_shape = (spimg.width, spimg.height)
-        img = Image.frombytes(pil_mode, img_shape, bytes(spimg.data))
+        img = imageIO.imageStructToPIL(spimg)
```

Comment:
One thing I do not understand looking at the implementation of `imageStructToPIL` is that it allegdly returns a `RGB` image, but we later do a `BGR` -> `RGB` conversion. Can you comment in the source code where you expect images to be `BGR` or `RGB`? It is a bit hard to follow...

Reply:
That's a good point, I should add comments. 

PIL expects the array in RGB order while ImageSchema expects BGR. So to get PIL from image schema I have to first flip BGR to RGB and then to get image schema from PIL I have to flip it back. The general rule is image schema is BGR, everything else depends on the particular library (e.g. PIL expects RGB, some keras models expect RGB some BGR).